Source/Core/CommandLineOptions.cs

```diff
@@ -569,7 +566,77 @@ void ObjectInvariant2()
     public bool InstrumentWithAsserts = false;
     public string ProverPreamble { get; set; }= null;
     public bool WarnNotEliminatedVars = false;
-    public bool PruneFunctionsAndAxioms = false;
+    
+    public enum PruneMode { 
+      None, 
```

Comment:
I don't fully understand the algorithmic pruning option, so I can't easily provide the detailed description you ask for. I think the summary is that any reference between declarations, except for those in quantifiers with triggers, is considered bidirectional.

I would be okay with removing automatic pruning once Dafny uses UsesClauses pruning.

Reply:
Ok, removing Automatic seems desirable since currently there are multiple overlapping ways of doing the same thing.  Would removing Automatic also mean that :exclude_dep will be dropped?

@RustanLeino : Any objection to dropping Automatic once Dafny uses UsesClauses pruning.  What is the timeline on that move?

Reply:
> Would removing Automatic also mean that :exclude_dep will be dropped?

Yes

Reply:
I have no objection to dropping the Automatic mode and the `:exclude_dep` attribute. Instead, we would only support the new `uses` clauses.

Reply:
@shazqadeer can we merge this? I don't have a strong opinion on the name of the `UsesClauses` enum. I think the difference with Automatic pruning, is that it only automatically detects _outgoing_ edges in axioms, so something like `OnlyUseOutgoingAxiomEdges` could be the name.